### ISSUE TITLE
box_cox not working properly: `IndexError`

### ISSUE THREAD

**Issue by Oliversinn** [danmcgib/PRegress#10]
To replicate error use dataset at: https://www.kaggle.com/datasets/lashagoch/life-expectancy-who-updated/data

 ```python
import pandas as pd
import pregress as pr

df = pd.read_csv("Life-Expectancy-Data-Updated.csv")

temp_reg = pr.fit("Life_expectancy ~  BMI", df)
pr.box_cox(temp_reg)

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)

[<ipython-input-82-b819bec10b10>](https://localhost:8080/#) in <cell line: 3>()
      1 temp_reg = pr.fit("Life_expectancy ~  BMI", df)
      2 # Pregress: the function doesn't work with this case
----> 3 pr.box_cox(temp_reg)

[/usr/local/lib/python3.10/dist-packages/pregress/modeling/box_cox.py](https://localhost:8080/#) in box_cox(model)
     28     ci_lambdas = lambdas[np.array(log_likelihood) >= ci_cutoff]
     29     lambda_lower = ci_lambdas[0]
---> 30     lambda_upper = ci_lambdas[-1]
     31 
     32     # Plot the log-likelihood as a function of lambda

IndexError: index 0 is out of bounds for axis 0 with size 0
```

Same error for:

```python
temp_reg = pr.fit("Life_expectancy ~  BMI", df)
pr.box_cox(temp_reg)

temp_reg = pr.fit("Life_expectancy ~  GDP_per_capita", df)
pr.box_cox(temp_reg)
```

**Comment by danielmcgibney** [comment]
In your case, your best lambda is 3.48 which will cause errors on some Box-Cox functions.  We will include error handling for box_cox in the next release of PRegress and allow lambda values outside of -2 and 2.

Best regards,
Daniel McGibney Ph.D.
Associate Professor of Professional Practice
***@***.***
Department of Management Science
5250 University Dr. | Kosar-Epstein 416
Coral Gables, FL 33146
Tel: (305) 284-6599 | Fax: (305) 284-2321


From: Oliver Mazariegos ***@***.***>
Date: Wednesday, September 25, 2024 at 11:34 PM
To: danmcgib/PRegress ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXTERNAL] [danmcgib/PRegress] box_cox not working properly: `IndexError` (Issue #10)
CAUTION: This email originated from outside the organization. DO NOT CLICK ON LINKS or OPEN ATTACHMENTS unless you know and trust the sender.

To replicate error use dataset at: https://www.kaggle.com/datasets/lashagoch/life-expectancy-who-updated/data<https://urldefense.com/v3/__https:/www.kaggle.com/datasets/lashagoch/life-expectancy-who-updated/data__;!!KVu0SnhVq1hAFvslES2Y!P1FqCBD-S-IkW8ANRCpenfZBkbEiNNmxFmuiVXg-XseK-3qY1YMfH9ZkrgnMJ2jj56qCAwAJfHUZHWm8LWuD_Cr8KilAxA$>

import pandas as pd

import pregress as pr



df = pd.read_csv("Life-Expectancy-Data-Updated.csv")



temp_reg = pr.fit("Life_expectancy ~  BMI", df)

pr.box_cox(temp_reg)



---------------------------------------------------------------------------

IndexError                                Traceback (most recent call last)



[<ipython-input-82-b819bec10b10>](https://localhost:8080/#) in <cell line: 3>()

     1 temp_reg = pr.fit("Life_expectancy ~  BMI", df)

     2 # Pregress: the function doesn't work with this case

----> 3 pr.box_cox(temp_reg)



[/usr/local/lib/python3.10/dist-packages/pregress/modeling/box_cox.py](https://localhost:8080/#) in box_cox(model)

    28     ci_lambdas = lambdas[np.array(log_likelihood) >= ci_cutoff]

    29     lambda_lower = ci_lambdas[0]

---> 30     lambda_upper = ci_lambdas[-1]

    31

    32     # Plot the log-likelihood as a function of lambda



IndexError: index 0 is out of bounds for axis 0 with size 0

Same error for:

temp_reg = pr.fit("Life_expectancy ~  BMI", df)

pr.box_cox(temp_reg)



temp_reg = pr.fit("Life_expectancy ~  GDP_per_capita", df)

pr.box_cox(temp_reg)

**Comment by danielmcgibney** [comment]
Issue resolved with release of 1.0.3.
